src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -631,31 +634,26 @@ public static CommandElement optionalWeak(CommandElement element, Object value)
 
         @Override
         public void parse(CommandSource source, CommandArgs args, CommandContext context) throws ArgumentParseException {
-            if (!args.hasNext()) {
-                Text key = this.element.getKey();
-                if (key != null && this.value != null) {
-                    context.putArg(key.toPlain(), this.value);
-                }
-                return;
-            }
-            Object startState = args.getState();
-            try {
-                this.element.parse(source, args, context);
-            } catch (ArgumentParseException ex) {
-                if (this.considerInvalidFormatEmpty || args.hasNext()) { // If there are more args, suppress. Otherwise, throw the error
-                    args.setState(startState);
-                    if (this.element.getKey() != null && this.value != null) {
-                        context.putArg(this.element.getUntranslatedKey(), this.value);
+            if (args.hasNext()) {
+                Object start = args.getState();
+                try {
+                    this.element.parse(source, args, context);
+                } catch (ArgumentParseException ex) {
+                    if (!this.considerInvalidFormatEmpty && !args.hasNext()) {
+                        throw ex;
                     }
-                } else {
-                    throw ex;
+                    args.setState(start);
                 }
             }
+            if (this.element.getKey() != null && this.value != null) {
+                context.putArg(this.element.getUntranslatedKey(), this.value);
+            }
         }
 
+        @Nullable
         @Override
         protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
-            return args.hasNext() ? null : this.element.parseValue(source, args);
+            return args.hasNext() ? this.element.parseValue(source, args) : this.value;
```

Comment:
This is a gross simplification over what it should be, as really it should be catching any thrown exception to have the same process as in `#parse`. I would personally prefer having things like this throw an `UnsupportedOperationException`, but it's too far along to change that now. This just brings it close to the expected behavior as it shouldn't be returning `null` if it has a next argument.

Reply:
No, my point is there was no point in changing this method because the method `parseValue`, a protected method in a private class so cannot be overriden on the outside, is never called. This probably _should_ throw an exception of some kind, but far better just to leave it and ignore it for the purposes of this PR because it's harmless.